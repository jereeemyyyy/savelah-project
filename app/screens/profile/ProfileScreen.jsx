import { Text, View } from 'react-native';
import { useEffect, useState } from 'react';
import { supabase } from '../../../lib/supabase';
import SignOutButton from './components/SignOutButton';
import Friends from './components/Friends';
import Extensions from './components/Extensions';
import Settings from './components/Settings';
import Profile from './components/Profile';

export default function ProfileScreen() {    

    const [userData, setUserData] = useState(null);
    
    useEffect(() => {
      const getUserData  = async () => {
        try {
          const { data: { user } } = await supabase.auth.getUser();
          setUserData(user);

        } catch (error) {
          console.error('Error getting user:', error.message);
        }
      }
      getUserData();
    }, []);
  
    return (
        <View className="flex-1 bg-gray-800 p-6" 
              testID="profileScreen"
        >
          <Profile userData={userData} 
                   testID="profileComponent"
          />
          <Text className="text-xl font-bold text-white m-1" testID="settingsTitle">Settings</Text>
          <Settings userData={userData} 
                    testID="settingsComponent"
          />
          <Text className="text-xl font-bold text-white m-1" testID="extensionsTitle">Extensions</Text>
          <Extensions testID="extensionsComponent"/>
          <Text className="text-xl font-bold text-white m-1" testID="friendsTitle">Friends</Text>
          <Friends testID="friendsComponent"/>
          <SignOutButton testID="signOutButton"/>
        </View>
    );
}
