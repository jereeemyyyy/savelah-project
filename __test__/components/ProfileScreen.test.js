import React from 'react';
import renderer from 'react-test-renderer';
import ProfileScreen from '../../app/screens/profile/ProfileScreen';

jest.mock('@react-navigation/native', () => {
    const actualNav = jest.requireActual('@react-navigation/native');
    return {
        ...actualNav,
        useNavigation: () => ({
            navigate: jest.fn(),
        }),
    };
});

jest.mock('expo-router', () => ({
    useNavigation: () => ({
        navigate: jest.fn(),
        reset: jest.fn(),
    }),
}));


describe('ProfileScreen', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<ProfileScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('contains the Profile component', () => {
        const component = renderer.create(<ProfileScreen />);
        const instance = component.root;

        expect(instance.findByProps({ testID: 'profileComponent' })).toBeTruthy();
    });

    it('contains the Settings title and component', () => {
        const component = renderer.create(<ProfileScreen />);
        const instance = component.root;

        expect(instance.findByProps({ testID: 'settingsTitle' }).props.children).toBe('Settings');
        expect(instance.findByProps({ testID: 'settingsComponent' })).toBeTruthy();
    });

    it('contains the Extensions title and component', () => {
        const component = renderer.create(<ProfileScreen />);
        const instance = component.root;

        expect(instance.findByProps({ testID: 'extensionsTitle' }).props.children).toBe('Extensions');
        expect(instance.findByProps({ testID: 'extensionsComponent' })).toBeTruthy();
    });

    it('contains the Friends title and component', () => {
        const component = renderer.create(<ProfileScreen />);
        const instance = component.root;

        expect(instance.findByProps({ testID: 'friendsTitle' }).props.children).toBe('Friends');
        expect(instance.findByProps({ testID: 'friendsComponent' })).toBeTruthy();
    });

    it('contains the SignOutButton component', () => {
        const component = renderer.create(<ProfileScreen />);
        const instance = component.root;

        expect(instance.findByProps({ testID: 'signOutButton' })).toBeTruthy();
    });
});
