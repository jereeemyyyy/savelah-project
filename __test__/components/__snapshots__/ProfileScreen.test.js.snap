// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileScreen renders correctly 1`] = `
<View
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "backgroundColor": "#1f2937",
      },
      {
        "paddingBottom": 24,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 24,
      },
    ]
  }
  testID="profileScreen"
>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "alignItems": "center",
        },
        {
          "marginBottom": 32,
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "width": 80,
            },
            {
              "height": 80,
            },
            {
              "backgroundColor": "#e5e7eb",
            },
            {
              "borderBottomLeftRadius": 9999,
              "borderBottomRightRadius": 9999,
              "borderTopLeftRadius": 9999,
              "borderTopRightRadius": 9999,
            },
            {
              "marginRight": 16,
            },
            {
              "alignItems": "center",
            },
            {
              "flexDirection": "row",
            },
            {
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text />
      </View>
    </View>
    <View>
      <Text
        style={
          [
            {
              "fontSize": 24,
              "lineHeight": 32,
            },
            {
              "fontWeight": "700",
            },
            {
              "color": "#fff",
            },
          ]
        }
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
            "position": "absolute",
            "right": -20,
            "top": 16,
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
  <Text
    style={
      [
        {
          "fontSize": 20,
          "lineHeight": 28,
        },
        {
          "fontWeight": "700",
        },
        {
          "color": "#fff",
        },
        {
          "marginBottom": 4,
          "marginLeft": 4,
          "marginRight": 4,
          "marginTop": 4,
        },
      ]
    }
    testID="settingsTitle"
  >
    Settings
  </Text>
  <View
    style={
      [
        {
          "marginBottom": 16,
        },
        {
          "paddingTop": 16,
        },
        {
          "backgroundColor": "#fff",
        },
        {
          "borderBottomLeftRadius": 12,
          "borderBottomRightRadius": 12,
          "borderTopLeftRadius": 12,
          "borderTopRightRadius": 12,
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 16,
          "marginLeft": 8,
          "marginRight": 8,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "flexDirection": "row",
            },
            {
              "alignItems": "center",
            },
          ]
        }
      >
        <Text />
        <Text
          style={
            [
              {
                "color": "#000",
              },
              {
                "marginLeft": 20,
                "marginRight": 20,
              },
            ]
          }
        />
      </View>
      <Text />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 16,
          "marginLeft": 8,
          "marginRight": 8,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "flexDirection": "row",
            },
            {
              "alignItems": "center",
            },
          ]
        }
      >
        <Text />
        <Text
          style={
            [
              {
                "color": "#000",
              },
              {
                "marginLeft": 20,
                "marginRight": 20,
              },
            ]
          }
        />
      </View>
      <Text />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 16,
          "marginLeft": 8,
          "marginRight": 8,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "flexDirection": "row",
            },
            {
              "alignItems": "center",
            },
          ]
        }
      >
        <Text />
        <Text
          style={
            [
              {
                "color": "#000",
              },
              {
                "marginLeft": 20,
                "marginRight": 20,
              },
            ]
          }
        >
          Edit Bank Account
        </Text>
      </View>
      <Text />
    </View>
  </View>
  <Text
    style={
      [
        {
          "fontSize": 20,
          "lineHeight": 28,
        },
        {
          "fontWeight": "700",
        },
        {
          "color": "#fff",
        },
        {
          "marginBottom": 4,
          "marginLeft": 4,
          "marginRight": 4,
          "marginTop": 4,
        },
      ]
    }
    testID="extensionsTitle"
  >
    Extensions
  </Text>
  <View
    style={
      [
        {
          "marginBottom": 16,
        },
        {
          "paddingTop": 16,
        },
        {
          "backgroundColor": "#fff",
        },
        {
          "borderBottomLeftRadius": 12,
          "borderBottomRightRadius": 12,
          "borderTopLeftRadius": 12,
          "borderTopRightRadius": 12,
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 16,
          "marginLeft": 8,
          "marginRight": 8,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "flexDirection": "row",
            },
            {
              "alignItems": "center",
            },
          ]
        }
      >
        <Text />
        <Text
          style={
            [
              {
                "color": "#000",
              },
              {
                "marginLeft": 20,
                "marginRight": 20,
              },
            ]
          }
        >
          Expenses Splitter
        </Text>
      </View>
      <Text />
    </View>
  </View>
  <Text
    style={
      [
        {
          "fontSize": 20,
          "lineHeight": 28,
        },
        {
          "fontWeight": "700",
        },
        {
          "color": "#fff",
        },
        {
          "marginBottom": 4,
          "marginLeft": 4,
          "marginRight": 4,
          "marginTop": 4,
        },
      ]
    }
    testID="friendsTitle"
  >
    Friends
  </Text>
  <View
    style={
      [
        {
          "marginBottom": 16,
        },
        {
          "paddingTop": 16,
        },
        {
          "backgroundColor": "#fff",
        },
        {
          "borderBottomLeftRadius": 12,
          "borderBottomRightRadius": 12,
          "borderTopLeftRadius": 12,
          "borderTopRightRadius": 12,
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 16,
          "marginLeft": 8,
          "marginRight": 8,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "flexDirection": "row",
            },
            {
              "alignItems": "center",
            },
          ]
        }
      >
        <Text />
        <Text
          style={
            [
              {
                "color": "#000",
              },
              {
                "marginLeft": 20,
                "marginRight": 20,
              },
            ]
          }
        >
          Add a Friend
        </Text>
      </View>
      <Text />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 16,
          "marginLeft": 8,
          "marginRight": 8,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "flexDirection": "row",
            },
            {
              "alignItems": "center",
            },
          ]
        }
      >
        <Text />
        <Text
          style={
            [
              {
                "color": "#000",
              },
              {
                "marginLeft": 20,
                "marginRight": 20,
              },
            ]
          }
        >
          Referral
        </Text>
      </View>
      <Text />
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#ef4444",
        "borderBottomLeftRadius": 9999,
        "borderBottomRightRadius": 9999,
        "borderTopLeftRadius": 9999,
        "borderTopRightRadius": 9999,
        "marginTop": 12,
        "opacity": 1,
        "paddingBottom": 8,
        "paddingTop": 8,
      }
    }
  >
    <Text
      style={
        [
          {
            "textAlign": "center",
          },
          {
            "color": "#fff",
          },
          {
            "fontWeight": "700",
          },
          {
            "fontSize": 18,
            "lineHeight": 28,
          },
        ]
      }
    >
      Sign Out
    </Text>
  </View>
</View>
`;
